### ISSUE TITLE
Unable to "mv" files greater than 5GB

### ISSUE THREAD

**Issue by rkeene** [s3tools/s3cmd#830]
Using s3cmd v1.6.0 (but no changes in 1.6.1 seem to address this), I cannot use "s3cmd mv <fileGreaterThan5GBInSize> <fileGreaterThan5GBInSize>,COMPLETE"

s3cmd exits with exit code 1 but does not print any diagnostics.

If I use --verbose, still nothing useful.

If I use --debug then I finally get to the cause of the error:
```
<Error>
    <Code>InvalidRequest</Code>
    <Message>The specified copy source is larger than the maximum allowable size for a copy source: 5368709120</Message>
    <RequestId>X</RequestId>
    <HostId>X</HostId
</Error>
```

Is there some other way to simply rename a file in the same bucket ?

**Comment by rkeene** [comment]
I confirmed that Pull Request #74 solves this (patch applied nearly cleanly against v1.6.0)

**Comment by surminus** [comment]
I recently came across this bug as I used s3cmd in some automated backup scripts that requires renaming of files. 

My solution in the end has been to use [`awscli`](https://aws.amazon.com/cli/) since they are capable of `mv`s against files greater than 5GB.

**Comment by nchammas** [comment]
The root cause is the same as for #818, I would guess.

**Comment by sn01615** [comment]
mv big files have no effect.
`s3cmd mv --recursive s3://aa/mysql-backup/xxday/xx.2019-03-12 s3://aa/mysql-backup/xxprevious_day/`

**Comment by bdoremus** [comment]
Been open for a while, but still an issue with upload sources from s3 greater than 5GB.
```
s3cmd  mv s3://file_source s3://file_destination -vd
```
```
DEBUG: S3Error: 400 (Bad Request)
DEBUG: HttpHeader: x-amz-request-id: --redacted--
DEBUG: HttpHeader: x-amz-id-2: --redacted--
DEBUG: HttpHeader: content-type: application/xml
DEBUG: HttpHeader: transfer-encoding: chunked
DEBUG: HttpHeader: date: Mon, 25 Nov 2019 17:17:13 GMT
DEBUG: HttpHeader: connection: close
DEBUG: HttpHeader: server: AmazonS3
DEBUG: ErrorXML: Code: 'InvalidRequest'
DEBUG: ErrorXML: Message: 'The specified copy source is larger than the maximum allowable size for a copy source: --redacted--'
DEBUG: ErrorXML: RequestId: '--redacted--'
DEBUG: ErrorXML: HostId: '--redacted--'
```

**Comment by fviard** [comment]
Good news, I finally had the time to rework completely the Multipart code to support properly the MultipartCopy, that was needed for all the copy/mv operations with files bigger than 5GB.

So, that fixed all the issues with sync, cp or mv of files bigger 5GB.
In addition I fixed a lot of small bugs, especially for error reporting, related to these features.

@surminus To be noted, we are now "more powerful" than aws cli because I think aws cli does not copy object metadata for big files (>8MB in the default config) but inconsistently does it for small files.

I also found a hack to support "modify" of metadata of objects >= 5GB, that I don't think any other cli tool has. Still, that will be a little bit slow, but I guess that it is better than not being able to do it. 

So everything is reader, tested, and pushed in the MASTER branch.
If you have the possibility to give it a try, that would be great.
Otherwise, it will be available in releases >= 2.2.0.
